### PR TITLE
[MaxText.inference_mlperf] pytype + pylint + pyink + codespell

### PR DESCRIPTION
**Pull request by SamuelMarks**
# Description

Many of my contributions have been to improve the overall quality of this codebase through application of automated code-quality tooling. You already have pre-commit hooks; this PR adds everything in your `code_style.sh` into it; so no longer will people be able to contribute code so low quality it doesn't pass the automated code-quality checking tooling.

NOTE: this is a subset of #1994 so should be merged before that PR.

# Tests

N/A

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/inference_mlperf/evaluate-accuracy-fast.py [hunk 1]
```diff
@@ -42,13 +42,13 @@ def split_data(preds: List[str], refs: List[str], num_chunks: int) -> List[Tuple
 
 
 def compute_rouge_chunk(chunk: Tuple[List[str], List[str]], metric) -> Dict:
-  """Compute ROUGE scores for a chunk of data"""
```
Comment (shralex) [inline MaxText/inference_mlperf/evaluate-accuracy-fast.py]: These fixes seem mistaken

**Review by shralex — CHANGES_REQUESTED** [review]

File: MaxText/inference_mlperf/evaluate-accuracy-fast.py (hunk 1 above)
Comment (SamuelMarks) [reply MaxText/inference_mlperf/evaluate-accuracy-fast.py]: Good find! - Ok fixed here and added to the `-L` of `codespell` in the `.pre-commit-config.yaml` of #1994

EDIT: In 770fcf7 Qinwen Wu added `gs://max-datasets-rogue` throughout the codebase. Should this be renamed also?

**Review by shralex — APPROVED** [review]

File: MaxText/inference_mlperf/evaluate-accuracy-fast.py (hunk 1 above)
Comment (suexu1025) [reply MaxText/inference_mlperf/evaluate-accuracy-fast.py]: is there any format mismatch? what's the recommended format?

Comment (SamuelMarks) [reply MaxText/inference_mlperf/evaluate-accuracy-fast.py]: @suexu1025 No, just spell checker issues

Comment (shralex) [reply MaxText/inference_mlperf/evaluate-accuracy-fast.py]: @suexu1025 should it be rouge instead of rogue :)

Comment (SamuelMarks) [reply MaxText/inference_mlperf/evaluate-accuracy-fast.py]: xmen-deal-with-it

File: MaxText/inference_mlperf/offline_inference.py [hunk 2]
```diff
@@ -353,7 +353,7 @@ def detokenize():
           self.params,
           self.decode_state,
           slot,
-          row.id,
+          int(row.id),
```
Comment (patemotter) [inline MaxText/inference_mlperf/offline_inference.py:356]: Is there something special about `row.id` that it needs to be explicitly cast to int but no other variables need the same treatment?

**Review by patemotter — APPROVED** [review]
Overall LGTM, just one minor Q on the int casting.

File: MaxText/inference_mlperf/offline_inference.py (hunk 2 above)
Comment (SamuelMarks) [reply MaxText/inference_mlperf/offline_inference.py:356]: @patemotter In `MaxText.inference_mlperf.offline_inference.PrefillHelper`, there's a `process` function. Its 4th parameter is of `int` type. Type errors were detected by `pytype` for having the wrong type here.

```py
  def process(
      self,
      model_params: Params,
      decode_state: DecodeState,
      decode_slot: int,
      input_id: int,
      input_tokens_padded: jax.Array,
      input_true_length: int,
      max_length: int,
      prefill_done: Callable[[List[Tuple[engine_api.ResultTokens, int]], List[int], DecodeState], None],
      rng: PRNGKeyType,
  ) -> None:
```

https://github.com/AI-Hypercomputer/maxtext/blob/347db31/MaxText/inference_mlperf/offline_inference.py#L129

**Review by bvandermoon — APPROVED** [review]
